free/src/main/scala/cats/free/Free.scala

```diff
@@ -250,3 +250,40 @@ object Free {
       def flatMap[A, B](a: Free[S, A])(f: A => Free[S, B]): Free[S, B] = a.flatMap(f)
     }
 }
+
+private trait FreeFoldable[F[_]] extends Foldable[Free[F,?]] {
+
+  implicit def F: Foldable[F]
+  implicit def FunctorF: Functor[F]
+
+  override final def foldLeft[A,B](fa: Free[F,A], b: B)(f: (B,A) => B): B =
+    fa.fold(f(b,_), F.foldLeft(_,b)((bb,freeA) => foldLeft(freeA,bb)(f)))
+
+  override final def foldRight[A,B](fa: Free[F,A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+    fa.fold(f(_, lb), F.foldRight(_,lb)( (freeA,eb) => foldRight(freeA,eb)(f)))
+}
+
+private trait FreeTraverse[F[_]] extends Traverse[Free[F,?]] with FreeFoldable[F] {
+  implicit def F: Traverse[F]
+
+  override final def traverse[G[_], A, B](fa: Free[F,A])(f: A => G[B])(implicit G: Applicative[G]): G[Free[F,B]] =
+    fa.fold(
+      a => G.map(f(a))(Free.pure(_)),
+      ffreeA => G.map(F.traverse(ffreeA)(traverse(_)(f)))(Free.roll(_))
+    )
+}
+
+sealed private[free] abstract class FreeInstances {
+
+  implicit def catsFoldableForFree[F[_] : Functor : Foldable]: Foldable[Free[F,?]] =
+    new FreeFoldable[F] {
+      def F = implicitly
+      def FunctorF = implicitly
+    }
+
+  implicit def catsTraverseForFree[F[_] : Functor : Traverse]: Traverse[Free[F, ?]] =
```

Comment:
`Traverse` is a `Functor` as well, so you can drop the `Functor` requirement.

I think we generally prefer "explicit implicits" (is that a thing?), compared to context bounds.